### PR TITLE
hamming.py

### PR DESCRIPTION
**Pull request by amymhaddad**
Hamming exercise and its test are attached. As discussed, I didn't include the "utility functions" from the test.

### CODE REVIEW

**Comment by amymhaddad** [comment]
@paulghaddad Great suggestions for shortening the code!

But I have a question:
The first time I did this problem I used range:
```

count = 0
for i in range(len(strand_a)):
        if strand_a[i] != strand_b[i]:
            count += 1

        return count
```
Pylint said I should use enumerate, so I did:

```
mismatch = []
for i, strand in enumerate(strand_a):
        if strand_a[i] != strand_b[i]:
            mismatch.append(strand_a[i])

        return len(mismatch)
```
Here's the question: the above code results in an error if I don't use "strand" (ie, `for i, strand in enumerate(strand_a):`). At the same time, pylint says "strand" is unused. What's going on here?!

File: hamming.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Tests to see if values of hamming strands are different"""
+
+def distance(strand_a, strand_b):
+    """Tests to find hamming distance"""
+
+    mismatch = []
+
+    length_a, length_b = len(strand_a), len(strand_b)
+
+    if length_a != length_b:
```
Comment (paulghaddad) [inline hamming.py]: Nice concise check!

File: hamming.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        raise ValueError("The lengths of the two strands are not the same")
+
+    if strand_a == '' or strand_b == '':
+        return 0
+
+    for i, strand in enumerate(strand_a):
+        if strand_a[i] != strand_b[i]:
+            mismatch.append(strand_a[i])
+
+        return len(mismatch)
```
Comment (paulghaddad) [inline hamming.py]: So I'm surprised this works...this last line is indented one too far, and has the effect of exiting on the first iteration of the `for` loop.

**Comment by paulghaddad** [comment]
@amymhaddad 

Good question. This commonly occurs where you don't use one of the variables. The convention is to replace the variable with an underscore instead. This tells the reader that it's unused:

```
for i, _ in enumerate(strand_a):
```

**Comment by amymhaddad** [comment]
@paulghaddad This exercise is updated!

**Pull request merged**